File: src/erlang/ch01/1.3/l0/src/divisibility.erl
-module(divisibility).
-export([is_divisible/2, last_decimal_digit/1, divisible_by_10/1]).

is_divisible(_A, 0) -> false;  %% undefined; keep simple at L0
is_divisible(A, B) when is_integer(A), is_integer(B) ->
    A rem B =:= 0.

last_decimal_digit(N) when is_integer(N) ->
    abs(N) rem 10.

divisible_by_10(N) ->
    last_decimal_digit(N) =:= 0.

File: src/go/ch01/1.3/l0/divisibility.go
package main

func IsDivisible(a, b int) bool {
	if b == 0 {
		return false // undefined; keep it simple for L0
	}
	return a%b == 0
}

func LastDecimalDigit(n int) int {
	if n < 0 {
		n = -n
	}
	return n % 10
}

func DivisibleBy10(n int) bool {
	return LastDecimalDigit(n) == 0
}
